From 9c3cb5a72a6593dabf4a1e62263e449d07eb5a6f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:13:13 -0400
Subject: [alsa] hda/hdmi: apply all Haswell fix-ups to Broadwell display codec

Message-id: <140972823573-25-patchset-jkysela@redhat.com>
Patchwork-id: 90725
O-Subject: [RHEL 7.1 PATCH 024/442] ALSA: hda/hdmi - apply all Haswell fix-ups to Broadwell display codec
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Broadwell and Haswell have the same behavior on display audio. So this patch
defines is_haswell_plus() to include codecs for both Haswell and its successor
Broadwell, and apply all Haswell fix-ups to Broadwell.

Signed-off-by: Mengdong Lin <mengdong.lin@intel.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 75dcbe4dc251ebc28cdf0797b85774cdf53a4d29)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_hdmi.c b/sound/pci/hda/patch_hdmi.c
index 670da3e..64f0a5e 100644
--- a/sound/pci/hda/patch_hdmi.c
+++ b/sound/pci/hda/patch_hdmi.c
@@ -46,6 +46,9 @@ module_param(static_hdmi_pcm, bool, 0644);
 MODULE_PARM_DESC(static_hdmi_pcm, "Don't restrict PCM parameters per ELD info");
 
 #define is_haswell(codec)  ((codec)->vendor_id == 0x80862807)
+#define is_broadwell(codec)    ((codec)->vendor_id == 0x80862808)
+#define is_haswell_plus(codec) (is_haswell(codec) || is_broadwell(codec))
+
 #define is_valleyview(codec) ((codec)->vendor_id == 0x80862882)
 
 struct hdmi_spec_per_cvt {
@@ -1101,7 +1104,7 @@ static void hdmi_setup_audio_infoframe(struct hda_codec *codec,
  if (!channels)
   return;
 
- if (is_haswell(codec))
+ if (is_haswell_plus(codec))
   snd_hda_codec_write(codec, pin_nid, 0,
          AC_VERB_SET_AMP_GAIN_MUTE,
          AMP_OUT_UNMUTE);
@@ -1280,7 +1283,7 @@ static int hdmi_setup_stream(struct hda_codec *codec, hda_nid_t cvt_nid,
  struct hdmi_spec *spec = codec->spec;
  int err;
 
- if (is_haswell(codec))
+ if (is_haswell_plus(codec))
   haswell_verify_D0(codec, cvt_nid, pin_nid);
 
  err = spec->ops.pin_hbr_setup(codec, pin_nid, is_hbr_format(format));
@@ -1421,7 +1424,7 @@ static int hdmi_pcm_open(struct hda_pcm_stream *hinfo,
        mux_idx);
 
  /* configure unused pins to choose other converters */
- if (is_haswell(codec) || is_valleyview(codec))
+ if (is_haswell_plus(codec) || is_valleyview(codec))
   intel_not_share_assigned_cvt(codec, per_pin->pin_nid, mux_idx);
 
  snd_hda_spdif_ctls_assign(codec, pin_idx, per_cvt->cvt_nid);
@@ -1611,7 +1614,7 @@ static int hdmi_add_pin(struct hda_codec *codec, hda_nid_t pin_nid)
  if (get_defcfg_connect(config) == AC_JACK_PORT_NONE)
   return 0;
 
- if (is_haswell(codec))
+ if (is_haswell_plus(codec))
   intel_haswell_fixup_connect_list(codec, pin_nid);
 
  pin_idx = spec->num_pins;
@@ -2249,7 +2252,7 @@ static int patch_generic_hdmi(struct hda_codec *codec)
  codec->spec = spec;
  hdmi_array_init(spec, 4);
 
- if (is_haswell(codec)) {
+ if (is_haswell_plus(codec)) {
   intel_haswell_enable_all_pins(codec, true);
   intel_haswell_fixup_enable_dp12(codec);
  }
@@ -2264,7 +2267,7 @@ static int patch_generic_hdmi(struct hda_codec *codec)
   return -EINVAL;
  }
  codec->patch_ops = generic_hdmi_patch_ops;
- if (is_haswell(codec)) {
+ if (is_haswell_plus(codec)) {
   codec->patch_ops.set_power_state = haswell_set_power_state;
   codec->dp_mst = true;
  }
-- 
1.7.1